hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -443,11 +443,11 @@ def _guess_strategy_by_argname(name: str) -> st.SearchStrategy:
     return st.nothing()
 
 
-def _get_params(func: Callable) -> Dict[str, inspect.Parameter]:
+def _get_params(func: Callable, eval_str: bool = False) -> Dict[str, inspect.Parameter]:
     """Get non-vararg parameters of `func` as an ordered dict."""
     var_param_kinds = (inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD)
     try:
-        params = list(get_signature(func).parameters.values())
+        params = list(get_signature(func, eval_str=eval_str).parameters.values())
     except Exception:
```

Comment:
```suggestion
    except Exception:
        if eval_str:
            return _get_params(func)
```

Evaluating string annotations can fail due to syntax errors, name resolution, or whatever arbitrary code execution we invoked failing; so in those cases I'd rather fall back to unevaluated annotations (above) or switching to `annotate=False` for that argument.  I trust your judgement as to which is preferable.

In either case, a test demonstrating that would be nice - say `f(a: "invalid ::: syntax", b: "1/0")`.